src/ert/analysis/_es_update.py

```diff
@@ -569,76 +567,38 @@ def analysis_ES(
-
-                    for param_correlation_set in param_correlation_sets:
-                        # Find the rows matching the parameter group
-                        matching_rows = np.all(c_bool == param_correlation_set, axis=1)
-                        # Get the indices of the matching rows
-                        row_indices = np.where(matching_rows)[0]
-                        X_chunk = temp_storage[param_group.name][param_batch_idx, :][
-                            row_indices, :
-                        ]
-                        S_chunk = S[param_correlation_set, :]
-                        observation_errors_loc = observation_errors[
-                            param_correlation_set
-                        ]
-                        observation_values_loc = observation_values[
-                            param_correlation_set
-                        ]
-                        smoother.fit(
-                            S_chunk,
-                            observation_errors_loc,
-                            observation_values_loc,
-                            noise=noise[param_correlation_set],
-                            truncation=truncation,
-                            inversion=ies.InversionType(module.ies_inversion),
-                            param_ensemble=param_ensemble,
-                        )
-                        temp_storage[param_group.name][
-                            param_batch_idx[row_indices], :
-                        ] = smoother.update(X_chunk)
 
             else:
-                smoother.fit(
-                    S,
-                    observation_errors,
-                    observation_values,
-                    noise=noise,
-                    truncation=truncation,
-                    inversion=ies.InversionType(module.ies_inversion),
-                    param_ensemble=param_ensemble,
-                )
+                # Use low-level ies API to allow looping over parameters
                 if active_indices := param_group.index_list:
-                    temp_storage[param_group.name][active_indices, :] = smoother.update(
-                        temp_storage[param_group.name][active_indices, :]
+                    # The batch of parameters
+                    X_local = temp_storage[param_group.name][active_indices, :]
+
+                    # Update manually using global transition matrix T
+                    temp_storage[param_group.name][active_indices, :] = (
+                        X_local + X_local @ T
                     )
+
                 else:
-                    temp_storage[param_group.name] = smoother.update(
-                        temp_storage[param_group.name]
-                    )
+                    # The batch of parameters
+                    X_local = temp_storage[param_group.name]
+
+                    # Update manually using global transition matrix T
+                    temp_storage[param_group.name] = X_local + X_local @ T
```

Comment:
nitpick: computing $X(I + T)$ is probably a bit faster than $X + X T$. the first one takes $nN + nN^2$ time, where $n$ is the number of parameters (rows) and $N$ the number of realizations (columns). the second one takes $N + nN^2$, if you compute $T + I$ by adding to the diagonal in place. please do double check this math :)

Reply:
Thanks, fixed.